test/optimizer/date_trunc_simplification_icu.test

```diff
@@ -0,0 +1,84 @@
+# name: test/optimizer/date_trunc_simplification_icu.test
+# description: test DATE_TRUNC() constant simplifications that require icu
+# group: [optimizer]
+
+require icu
+
+statement ok
+PRAGMA enable_verification;
```

Comment:
Heh you also need to set the calendar:

```sql
set Calendar='gregorian';
```
Otherwise if the testing locale's default calendar is not Gregorian, the tests will fail.  We set it for CI, but I've talked to people from China who had a different default calendar(!)